<div style="background:linear-gradient(135deg,#042f2e 0%,#0f766e 55%,#14b8a6 100%);border-radius:18px;padding:34px 30px;color:#fff;font-family:Inter,Segoe UI,sans-serif">
  <div style="font-size:12px;letter-spacing:3px;color:#99f6e4;font-weight:700;text-transform:uppercase">Chapter 37 · Solutions</div>
  <div style="font-size:34px;font-weight:900;line-height:1.1;margin:10px 0 6px">Discrete Distributions, Worked Answers ✅</div>
  <div style="font-size:15px;color:#d1fae5;max-width:720px;line-height:1.6">Full solutions to the five challenges. Try them yourself first, then compare.</div>
  <div style="margin-top:16px;font-size:13px;color:#a7f3d0">Statistics, Data Science and AI: A Visual Handbook · John Fisher · 2026</div>
</div>

## ⚙️ Setup

In [1]:
import numpy as np
from scipy import stats
rng = np.random.default_rng(371)
print("ready")

ready


<div style="background:#ecfdf5;border-left:5px solid #0d9488;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0d9488;letter-spacing:1px">CHALLENGE 1 · BINOMIAL</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Exactly 3 sixes in 5 rolls</div>
<div style="color:#4a5578;margin-top:6px">Roll a fair die 5 times. Find P(exactly 3 sixes) and the expected number of sixes.</div>
</div>

In [2]:
n, p = 5, 1/6
P3 = stats.binom.pmf(3, n, p)
print(f"P(exactly 3 sixes) = C(5,3)(1/6)^3(5/6)^2 = {P3:.4f}")
print(f"expected sixes = np = {n*p:.4f}")

P(exactly 3 sixes) = C(5,3)(1/6)^3(5/6)^2 = 0.0322
expected sixes = np = 0.8333


**Answer:** P(exactly 3 sixes) = C(5,3)(1/6)&sup3;(5/6)&sup2; &approx; **0.0322**, about 3%. The expected number of sixes is np = 5/6 &approx; 0.83. Three sixes in five rolls is well above average, hence uncommon.

<div style="background:#ecfdf5;border-left:5px solid #0d9488;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0891b2;letter-spacing:1px">CHALLENGE 2 · GEOMETRIC</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Waiting for the first success</div>
<div style="color:#4a5578;margin-top:6px">Each attempt succeeds with probability p = 0.25. Find the expected number of attempts to the first success, and P(first success on attempt 3).</div>
</div>

In [3]:
p = 0.25
print(f"E[attempts] = 1/p = {1/p:.1f}")
P3 = stats.geom.pmf(3, p)
print(f"P(first success on attempt 3) = (0.75)^2 (0.25) = {P3:.4f}")

E[attempts] = 1/p = 4.0
P(first success on attempt 3) = (0.75)^2 (0.25) = 0.1406


**Answer:** the expected wait is 1/p = **4** attempts. P(first success on attempt 3) = (1&minus;p)&sup2; p = (0.75)&sup2;(0.25) &approx; **0.1406**: you must fail twice, then succeed.

<div style="background:#ecfdf5;border-left:5px solid #0d9488;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0d9488;letter-spacing:1px">CHALLENGE 3 · POISSON</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">A busy call center</div>
<div style="color:#4a5578;margin-top:6px">Calls arrive at an average rate of 4 per hour. Using the Poisson distribution, find P(exactly 2 calls in an hour) and P(no calls).</div>
</div>

In [4]:
lam = 4
print(f"P(X=2) = e^-4 * 4^2 / 2! = {stats.poisson.pmf(2, lam):.4f}")
print(f"P(X=0) = e^-4           = {stats.poisson.pmf(0, lam):.4f}")

P(X=2) = e^-4 * 4^2 / 2! = 0.1465
P(X=0) = e^-4           = 0.0183


**Answer:** P(exactly 2) = e&#8315;&#8308;(4&sup2;/2!) &approx; **0.1465**, and P(no calls) = e&#8315;&#8308; &approx; **0.0183**. With a mean of 4, both 2 calls and 0 calls are below average, so 0 is quite unlikely.

<div style="background:#ecfdf5;border-left:5px solid #0d9488;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#db2777;letter-spacing:1px">CHALLENGE 4 · HYPERGEOMETRIC</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Quality control without replacement</div>
<div style="color:#4a5578;margin-top:6px">A box has 50 parts, 5 of them defective. You draw 10 without replacement. Find P(exactly 1 defective).</div>
</div>

In [5]:
M, K, N = 50, 5, 10
P1 = stats.hypergeom.pmf(1, M, K, N)
print(f"P(exactly 1 defective) = {P1:.4f}")
print(f"expected defectives = N*K/M = {N*K/M:.2f}")

P(exactly 1 defective) = 0.4313
expected defectives = N*K/M = 1.00


**Answer:** P(exactly 1 defective) &approx; **0.431**. The expected number of defectives is N&middot;K/M = 10(5/50) = 1.0. Because the draws are without replacement, this is hypergeometric, not binomial, though with such a small sample the two are close.

<div style="background:#ecfdf5;border-left:5px solid #0d9488;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0f766e;letter-spacing:1px">CHALLENGE 5 · MAXIMUM LIKELIHOOD</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Estimating p from data</div>
<div style="color:#4a5578;margin-top:6px">A classifier is tested on 20 cases and is correct on 8. What is the maximum-likelihood estimate of its success probability p?</div>
</div>

In [6]:
k, n = 8, 20
p_hat = k/n
print(f"MLE p_hat = k/n = {p_hat:.2f}")
# confirm: log-likelihood is maximized at the sample proportion
grid = np.linspace(0.01,0.99,400)
loglik = k*np.log(grid)+(n-k)*np.log(1-grid)
print(f"argmax of log-likelihood on a grid = {grid[np.argmax(loglik)]:.2f}")

MLE p_hat = k/n = 0.40
argmax of log-likelihood on a grid = 0.40


**Answer:** the MLE is p&#770; = k/n = 8/20 = **0.40**. For Bernoulli/binomial data the maximum-likelihood estimate of p is simply the observed proportion of successes, which is exactly the quantity a logistic-regression classifier is trained to predict.

---
<div style="background:#ffffff;border:1px solid #e6e9f2;border-radius:16px;padding:22px 26px;font-family:Inter,sans-serif">
<div style="font-size:19px;font-weight:800;color:#1a2138">🎉 Nicely done</div>
<div style="color:#4a5578;line-height:1.8;margin-top:8px">You used the binomial, geometric, Poisson, and hypergeometric distributions to answer real counting questions, and estimated a probability by maximum likelihood. Recognizing which distribution fits a problem is half the battle, the formulas follow from the recipe.</div>
</div>

<div style="text-align:center;color:#8b94b3;font-size:12px;margin-top:16px">Statistics, Data Science and AI: A Visual Handbook · © 2026 John Fisher</div>